### PR TITLE
Get rid of the mypy wrapper script

### PR DESCRIPTION
**Pull request by wsanchez**
Get rid of the mypy script, ignore untyped definitions in a bunch of files, and apply enough `# type: ignore hooey` to quiet various errors from `mypy` for now.

Note that number `# type: ignore[misc]` are added to silence errors related to `zope.interface` classes and a work-around using `Union` types from @glyph.

This also removed unnecessary references to type symbols that were added to silence `flake8`, which now sees uses in type hints, and then removes some actually unused symbols that are caught by `flake8`.

### CODE REVIEW

**Comment by wsanchez** [comment]
@glyph: done

**Comment by glyph** [comment]
It looks like I reviewed & merged the dependency-upgrade PR first and that has created conflicts again 😭

**Comment by wsanchez** [comment]
@glyph: extra-super-done

File: src/klein/_dihttp.py [hunk 1]
```diff
@@ -49,7 +47,7 @@ def urlFromRequest(request):
     )
 
 
-@provider(IRequiredParameter, IDependencyInjector)
+@provider(IRequiredParameter, IDependencyInjector)  # type: ignore[misc]
```
Comment (glyph) [inline src/klein/_dihttp.py]: When we do the ZI mypy plugin thing, will it obviate (and start warning on) all these `ignore[misc]`s? I want to make sure we can find these again easily to clean them up, and normally I’d say “drop a comment in there” but if we don’t need a comment then great.

File: src/klein/_isession.py [hunk 2]
```diff
@@ -15,12 +15,11 @@
...
     from twisted.internet.defer import Deferred
     from twisted.python.components import Componentized
-    from typing import Dict, Iterable, List, Text, Type, Sequence
+    from typing import Dict, Iterable, Text, Sequence
     from twisted.web.iweb import IRequest
     from zope.interface.interfaces import IInterface
 
-    Deferred, Text, Componentized, Sequence, IRequest, List, Type
-    Iterable, IInterface, Dict
+
```
Comment (glyph) [inline src/klein/_isession.py]: Huh, how are we getting rid of these? Has pyflakes just upped its game recently? (Very cool to do so, though…)

File: src/klein/interfaces.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 )
 
 if TYPE_CHECKING:               # pragma: no cover
-    from ._storage.memory import MemorySessionStore, MemorySession
+    from .storage._memory import MemorySessionStore, MemorySession
```
Comment (glyph) [inline src/klein/interfaces.py]: Ugh this  double-private thing just reminds me we really need to get `publication` in here at some point (not really germane to this change though).

File: tox.ini [hunk 4]
```diff
@@ -98,11 +98,12 @@ deps =
     flake8-import-order==0.18.1
     flake8-mutable==1.2.0
     flake8-pep3101==1.2.1
-    flake8==3.7.9
     mccabe==0.6.1
     pep8-naming==0.8.2
     pycodestyle==2.5.0
     pydocstyle==4.0.1
+    # pin pyflakes pending a release with https://github.com/PyCQA/pyflakes/pull/455
```
Comment (glyph) [inline tox.ini]: Thanks for the link for the reason for the pin!

File: tox.ini [hunk 5]
```diff
@@ -166,11 +167,13 @@ max-complexity = 60
 
 
 ##
-# Mypy linting
+# Mypy static type checking
```
Comment (glyph) [inline tox.ini]: ++

File: tox.ini [hunk 6]
```diff
@@ -182,24 +185,100 @@ deps =
...
 
 [mypy]
 
 # Global settings
 
-warn_redundant_casts = True
-warn_unused_ignores = True
-strict_optional = True
-show_column_numbers = True
+disallow_untyped_defs    = True
```
Comment (glyph) [inline tox.ini]: Somewhere in here we probably want `disallow_any_generics` but this is 100% not a blocker.

File: src/klein/_dihttp.py (hunk 1 above)
Comment (wsanchez) [reply src/klein/_dihttp.py]: I don't think we need a comment; unused ignored generate a warning.

In any case, I'm going to want to eventually search for all of the ignores and see about cleaning them up or filing `mypy` bugs.

File: src/klein/_isession.py (hunk 2 above)
Comment (wsanchez) [reply src/klein/_isession.py]: `pyflakes` has upped its game. (yay!)

File: src/klein/interfaces.py (hunk 3 above)
Comment (wsanchez) [reply src/klein/interfaces.py]: @glyph is talking about https://pypi.org/project/publication/, I think

File: tox.ini (hunk 6 above)
Comment (wsanchez) [reply tox.ini]: added to the TODO list

File: src/klein/interfaces.py (hunk 3 above)
Comment (glyph) [reply src/klein/interfaces.py]: Yep, that’s the one.

**Pull request merged**